ElementX/Sources/Other/Pills/PillAttachmentViewProvider.swift

```diff
@@ -43,14 +40,15 @@ final class PillAttachmentViewProvider: NSTextAttachmentViewProvider {
             return
         }
         
-        let viewModel: PillViewModel
+        let viewModel: PillContext
         let imageProvider: ImageProviderProtocol?
-        if isXcodePreview || Tests.isRunningTests {
-            viewModel = PillViewModel.mockViewModel(type: .loadUser)
+        if ProcessInfo.isXcodePreview || ProcessInfo.isRunningTests {
+            // The mock viewModel simulates the loading logic for testing purposes
+            viewModel = PillContext.mockViewModel(type: .loadUser)
```

Comment:
Maybe drop the ViewModel on the mock too?
```suggestion
            viewModel = PillContext.mock(type: .loadUser)
```